### DISCUSSION TITLE
3rd party API

### DISCUSSION THREAD

**Discussion by juan-manuel-alberro** [blitz-js/blitz#2373] [Q&A]
Hello guys,

I have a question and I wasn't able to find the answer in the documentation, I need to call an external REST API, what's the best practice for those cases? Do I need to create a query (`/queries` folder and then `useQuery`) or do I need to create a new API (`/api` folder) and later `fetch` that endpoint? 
I believe both options will work but I want to understand the design pattern to follow in Blitz 😃 

Thanks.

**Comment by MrLeebo** [comment]
Either approach will work for accessing an external API. You can generally prefer queries if you just want some data and not to worry about the specifics of the HTTP requests, and you can use custom APIs if you need more control (such as a non-JSON content type, or special security headers)

Another approach is getServerSideProps if the data is specific to a page and is something that you need to be in the initial page render, like SEO meta tags.
